File: sdkwork-backend-sdk-typescript/src/api/ppt-template-rendering.ts
import { backendApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { PlusApiResult, PlusPptTemplateRenderForm } from '../types';


export class PptTemplateRenderingApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Render PPT template */
  async render(body: PlusPptTemplateRenderForm): Promise<PlusApiResult> {
    return this.client.post<PlusApiResult>(backendApiPath(`/ppt/template/render`), body);
  }
}

export function createPptTemplateRenderingApi(client: HttpClient): PptTemplateRenderingApi {
  return new PptTemplateRenderingApi(client);
}

File: sdkwork-backend-sdk-go/api/ppt_template_rendering.go
package api

import (
    sdktypes "github.com/sdkwork/backend-sdk/types"
    sdkhttp "github.com/sdkwork/backend-sdk/http"
)

type PptTemplateRenderingApi struct {
    client *sdkhttp.Client
}

func NewPptTemplateRenderingApi(client *sdkhttp.Client) *PptTemplateRenderingApi {
    return &PptTemplateRenderingApi{client: client}
}

// Render PPT template
func (a *PptTemplateRenderingApi) Render(body sdktypes.PlusPptTemplateRenderForm) (sdktypes.PlusApiResult, error) {
    raw, err := a.client.Post(BackendApiPath("/ppt/template/render"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResult
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResult](raw)
}
